autosklearn/pipeline/components/classification/mlp.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import UniformFloatHyperparameter, \
+    UniformIntegerHyperparameter, UnParametrizedHyperparameter, Constant, \
+    CategoricalHyperparameter
+
+from autosklearn.pipeline.components.base import (
+    AutoSklearnClassificationAlgorithm,
+    IterativeComponent,
+)
+from autosklearn.pipeline.constants import SPARSE, DENSE, UNSIGNED_DATA, PREDICTIONS
+from autosklearn.util.common import check_for_bool
+
+
+class MLPClassifier(
+    IterativeComponent,
+    AutoSklearnClassificationAlgorithm
+):
+    def __init__(self, hidden_layer_depth, num_nodes_per_layer, activation, alpha,
+                 learning_rate_init, early_stopping, solver, batch_size,
+                 # n_iter_no_change, validation_fraction=None, tol,
+                 shuffle, beta_1, beta_2, epsilon,
+                 random_state=None, verbose=0):
+        self.hidden_layer_depth = hidden_layer_depth
+        self.num_nodes_per_layer = num_nodes_per_layer
+        self.max_iter = self.get_max_iter()
+        self.activation = activation
+        self.alpha = alpha
+        self.learning_rate_init = learning_rate_init
+        self.early_stopping = early_stopping
+        # self.n_iter_no_change = n_iter_no_change
+        # self.validation_fraction = validation_fraction
+        # self.tol = tol
+        self.solver = solver
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.epsilon = epsilon
+        self.beta_1 = beta_1
+
+        self.random_state = random_state
+        self.verbose = verbose
+        self.estimator = None
+        self.fully_fit_ = False
```

Comment:
This might be an issue. `sklearn` uses a trailing underscore to determine that an algorithm has been fit. Maybe this should be a private variable.

Reply:
Sure, I replaced it with _fully_fit